panic: byte index 12 is not a char boundary

```
panic: byte index 12 is not a char boundary; it is inside '打' (bytes 10..13) of `crates/实打实打算/d's'd`
  File "panicking.rs", line 515, in rust_begin_unwind
  ?, in route_recognizer::nfa::Thread::extract
  ?, in route_recognizer::nfa::NFA<T>::process
  ?, in route_recognizer::Router<T>::recognize
  ?, in conduit_router::RouteBuilder::call
...
(32 additional frame(s) were not displayed)
```

this can be reproduced by:

```rust
#[test]
fn test_chinese() {
    let mut router = Router::new();
    router.add("/crates/:foo/:bar", "Hello".to_string());

    let m = router.recognize("/crates/实打实打算/d's'd").unwrap();
    assert_eq!(m.handler().as_str(), "Hello");
    assert_eq!(m.params().find("foo"), Some("实打实打算"));
    assert_eq!(m.params().find("bar"), Some("d's'd"));
}
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -571,4 +571,15 @@ mod tests {
             "Hello"
         );
     }
+
+    #[test]
+    fn test_chinese() {
+        let mut router = Router::new();
+        router.add("/crates/:foo/:bar", "Hello".to_string());
+
+        let m = router.recognize("/crates/实打实打算/d's'd").unwrap();
+        assert_eq!(m.handler().as_str(), "Hello");
+        assert_eq!(m.params().find("foo"), Some("实打实打算"));
+        assert_eq!(m.params().find("bar"), Some("d's'd"));
+    }
 }

--- a/src/nfa.rs
+++ b/src/nfa.rs
@@ -227,7 +227,7 @@ impl<T> NFA<T> {
     {
         let mut threads = vec![Thread::new()];
 
-        for (i, char) in string.chars().enumerate() {
+        for (i, char) in string.char_indices() {
             let next_threads = self.process_char(threads, char, i);
 
             if next_threads.is_empty() {

